app/src/main/java/de/moyapro/idleworldsim/domain/consumption/Resources.kt

```diff
@@ -1,65 +1,98 @@
 package de.moyapro.idleworldsim.domain.consumption
 
 import de.moyapro.idleworldsim.domain.Species
-import de.moyapro.idleworldsim.domain.consumption.Resource.*
+import de.moyapro.idleworldsim.domain.SpeciesConstants
+import de.moyapro.idleworldsim.domain.valueObjects.*
+import de.moyapro.idleworldsim.domain.valueObjects.ResourceType.*
+import java.util.*
+
 
 data class Resources(
-    val quantities: DoubleArray = DoubleArray(values().size) { if (it == EvolutionPoints.ordinal) 0.0 else 1000.0 },
-    val populations: MutableMap<Species, Double> = mutableMapOf()
+    val quantities: MutableMap<ResourceType, Double> = mutableMapOf(*(values().map { Pair(it, if (it == EvolutionPoints) 0.0 else 1000.0) }).toTypedArray()),
+    val populations: MutableMap<Species, Population> = mutableMapOf()
 ) {
-    constructor(quantitiesMap: Map<Resource, Double>) : this(mapToDoubleArray(quantitiesMap))
-
-    companion object {
-        private fun mapToDoubleArray(quantitiesMap: Map<Resource, Double>): DoubleArray {
-            return DoubleArray(values().size) { quantitiesMap[values()[it]] ?: 0.0 }
-        }
+    @Deprecated("Should use Resource value object instead")
+    constructor(quantitiesMap: Map<ResourceType, Double>) : this(quantitiesMap.toMutableMap())
+    constructor(resource: ResourceType, quantity: Double = 1.0) : this(quantities = mutableMapOf(Pair(resource, quantity)))
+    constructor(species: Species, population: Population) : this(mutableMapOf(), mutableMapOf(Pair(species, population)))
+    constructor(doubleArrayOf: DoubleArray) : this(doubleArrayOf.withIndex().map { Pair(ResourceType.values()[it.index], it.value) }.associate { it }.toMutableMap())
```

Comment:
Kürzer mit `mapIndexed`:
`this(doubleArrayOf.mapIndexed { key, value ->Pair(ResourceType.values()[index], value) }.associate { it }.toMutableMap())`
Aber noch besser wäre eine extension function alias `associateWithIndexed` 😃

Reply:
DoubleArray will ich eh loswerden. Hatte nur keine Lust allle Stellen im Test anzupassen. Aber kann ich anpassen

Reply:
es geht sogar so constructor(resources: DoubleArray) : this(resources.mapIndexed{i,value -> Resources(values()[i], value)})
allerding habe ich dann schon die instanz. ich brauche aber einen constructoraufruf.

Ich hab es jetzt so gemacht: 
constructor(resources: DoubleArray) : this(resources.mapIndexed{i,value -> Pair(values()[i], value)}.associate { it }.toMutableMap())